pkg/ebpf/printk_patcher.go

```diff
@@ -207,3 +207,22 @@ func patchPrintkInstructions(p *ebpf.ProgramSpec) (int, error) {
 
 	return numPatches, errors.Join(errs...)
 }
+
+// PrintkPatcherModifier adds an InstructionPatcher to the manager that removes the newline character from log_debug calls if needed
+type PrintkPatcherModifier struct {
+}
+
+func (t *PrintkPatcherModifier) String() string {
+	return "PrintkPatcherModifier"
+}
+
+// BeforeInit adds the PatchPrintkNewline function to the manager
+func (t *PrintkPatcherModifier) BeforeInit(m *manager.Manager, _ *manager.Options) error {
+	m.InstructionPatchers = append(m.InstructionPatchers, PatchPrintkNewline)
```

Comment:
nit: does `PatchPrintkNewline` need to be exported anymore?

Reply:
Yes, in case someone wants to use it directly instead of relying on the default modifiers.

Reply:
I'm not talking about the modifier, but the patching function itself

Reply:
Oh my mistake, sorry. That one doesn't need to be exported, I'll change it now.